src/Umbraco.Cms.Api.Management/Controllers/DataType/ReferencesDataTypeController.cs

```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Api.Management.Factories;
+using Umbraco.Cms.Api.Management.ViewModels.DataType;
+using Umbraco.Cms.Core;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Api.Management.Controllers.DataType;
+
+public class ReferencesDataTypeController : DataTypeControllerBase
+{
+    private readonly IDataTypeService _dataTypeService;
+    private readonly IDataTypeReferenceViewModelFactory _dataTypeReferenceViewModelFactory;
+
+    public ReferencesDataTypeController(IDataTypeService dataTypeService, IDataTypeReferenceViewModelFactory dataTypeReferenceViewModelFactory)
+    {
+        _dataTypeService = dataTypeService;
+        _dataTypeReferenceViewModelFactory = dataTypeReferenceViewModelFactory;
+    }
+
+    [HttpGet("{key:guid}/references")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(DataTypeReferenceViewModel[]), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<DataTypeViewModel>> References(Guid key)
+    {
+        IDataType? dataType = _dataTypeService.GetDataType(key);
+        if (dataType == null)
+        {
+            return NotFound();
+        }
+
+        IReadOnlyDictionary<Udi, IEnumerable<string>> usages = _dataTypeService.GetReferences(dataType.Id);
+        DataTypeReferenceViewModel[] viewModels = _dataTypeReferenceViewModelFactory.CreateDataTypeReferenceViewModels(usages).ToArray();
+
+        return await Task.FromResult(Ok(viewModels));
+    }
```

Comment:
@Zeegaan I think we need to leave this as-is (unpaged). The single list of usages span multiple types of usage, which will make it hard to build a UI like this on top of a paginated response:

![image](https://user-images.githubusercontent.com/7405322/210942556-7b89b419-32e8-48b5-84db-086051866f26.png)

If we *really* want pagination, we need to split the single "references" endpoint into three endpoints, one for each potential type of reference (content type, media type, member type)... but that really seems to be over complicating things, specially when the number of references are expected to be relatively low.

Thoughts?

Reply:
@kjac Hmm, I do see your point there, lets's leave it as is, as it will not be breaking to make 3 new breakpoints, so we can "always" do that if needed 👍